BitFaster.Caching/Lazy/Atomic.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BitFaster.Caching.Lazy
+{
+    // Should this be called simply Atomic?
+    // Then we have Atomic and Scoped, and ScopedAtomic
+    // Then AtomicAsync, ScopedAsync, and ScopedAtomicAsync would follow, but there is no ScopedAsync equivalent at this point. That should be IAsyncDisposable (.net Core 3.1 onwards). That would imply that GetOrAdd async understands IAsyncDispose
+
+    // https://github.com/dotnet/runtime/issues/27421
+    // https://github.com/alastairtree/LazyCache/issues/73
+    public class Atomic<T>
```

Comment:
Compare to Lazy:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Lazy.cs#L319

- Lazy stores only value, func, and a state object (size = 3 refs)
- This has 3 reference type members plus isInitialized
- value factory is not set to null when done - could leak the factory